core/common/src/main/java/alluxio/worker/block/io/BlockClient.java

```diff
@@ -11,10 +11,57 @@
 
 package alluxio.worker.block.io;
 
+import com.google.common.base.MoreObjects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
 
 /**
- * Base interface for block reader/writers.
+ * Provides basic tracking and representation for block reader/writer clients.
  */
-public interface BlockClient extends Closeable {
+public abstract class BlockClient implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(BlockClient.class);
+
+  /** Used to keep unique client ids. */
+  private static AtomicInteger sNextClientId = new AtomicInteger(0);
```

Comment:
I think this should also be `final` since it's the reference to the `AtomicInteger` object